packages/block-library/src/breadcrumbs/index.php

```diff
@@ -115,6 +143,125 @@ function block_core_breadcrumbs_get_hierarchical_post_type_breadcrumbs( $post_id
 }
 
 
+/**
+ * Generates breadcrumb items for archive pages.
+ *
+ * Handles taxonomy archives, post type archives, date archives, and author archives.
+ * For hierarchical taxonomies, includes ancestor terms in the breadcrumb trail.
+ *
+ * @since 6.9.0
+ *
+ * @return array Array of breadcrumb HTML items.
+ */
+function block_core_breadcrumbs_get_archive_breadcrumbs() {
+	$breadcrumb_items = array();
+
+	// Date archive (check first since it doesn't have a queried object).
+	if ( is_date() ) {
+		$year  = get_query_var( 'year' );
+		$month = get_query_var( 'monthnum' );
+		$day   = get_query_var( 'day' );
+
+		if ( $year ) {
+			if ( $month ) {
+				// Year is linked if we have month.
+				$breadcrumb_items[] = sprintf(
+					'<a href="%s">%s</a>',
+					esc_url( get_year_link( $year ) ),
+					esc_html( $year )
+				);
+
+				if ( $day ) {
+					// Month is linked if we have day.
+					$breadcrumb_items[] = sprintf(
+						'<a href="%s">%s</a>',
+						esc_url( get_month_link( $year, $month ) ),
+						esc_html( date_i18n( 'F', mktime( 0, 0, 0, $month, 1 ) ) )
```

Comment:
Paranoid question: is it at all possible, in a weird scenario, that `mktime( 0, 0, 0, $month, 1 )` produces a value different from `mktime( 0, 0, 0, $month, 1, $year )`? 😅 Probably not, but just putting it out there.

Reply:
This is such a `Miguel` comment 😆 . I'm not sure, but since we have the year already we can pass it to be safe and not risk any weird scenario from `mktime` using the current year as default.